### ISSUE TITLE
Missing Attribute landing content when selecting filter with Ajax enabled.

### ISSUE THREAD

**Issue by koentjeh** [EmicoEcommerce/Magento2AttributeLandingTweakwise-archived#10]
Hi,

I've created a landing page that shows content when the color blue is filtered.

When opening the layered navigation with AJAX enabled and no filters selected. Selecting the color blue does not load the landing page content. Reloading the page does show the content. The issue is that the content is not loaded by AJAX and requires a page reload.

**Comment by edwinljacobs** [comment]
Hi,

What do you mean with "I've created a landing page that shows content when the color blue is filtered." I assume that the color blue is used as a landingspage filter? As in the added screenshot
![blue-product-landingpage](https://user-images.githubusercontent.com/11278380/103672490-abb16200-4f7c-11eb-9da7-0cbc7676a039.png)

Or do you mean something else? Perhaps you have "hide selected filters" with the "off" value? Can you explain to me what I need to do to reproduce your issue?

With kind regards

**Comment by koentjeh** [comment]
You've assumed it right. I added a single filter attribute and filled it with some content which works as expected when navigating to https://example.com/category/color/blue.

![Schermafbeelding 2021-01-06 om 13 07 46](https://user-images.githubusercontent.com/6535718/103767107-54fd6400-5020-11eb-8833-048d610fe136.png)

![Schermafbeelding 2021-01-06 om 13 13 55](https://user-images.githubusercontent.com/6535718/103767533-17e5a180-5021-11eb-8a0b-bbb2bbf1b066.png)

![Schermafbeelding 2021-01-06 om 13 14 44](https://user-images.githubusercontent.com/6535718/103767583-2f248f00-5021-11eb-9312-f21b930788d1.png)

The result is still as expected when navigating to https://example.com/category/color/blue with Ajax filtering enabled in Store -> Configuration > Catalog > Tweakwise. However navigating to https://example.com/category/ and then selecting the color blue will reload the products with Ajax, but will not show/load the attribute content ("Als je dit ziet is de test geslaagd.", etc.) even though the color blue filter is selected. Reloading the page (which keeps the same blue filter active) does show the content.

**Translation**
kleur = color
blauw = blue

**Comment by edwinljacobs** [comment]
Ah ok,

There is a setting for that, it can be found under Stores > Configuration > Emico Extensions > Attribute Landing pages > Allow crosslink. This should have value "yes" for this to work. Each landingpage also has the option to override this setting via "Allow link in faceted search".

Hope this helps

**Comment by koentjeh** [comment]
I believe you misunderstood me. The issue is that this module misses the functionality to load the attribute landing content with Ajax while using the layered navigation on the category page with Ajax enabled.

Is it true and does this module not have this functionality or is it a valid issue?

**Comment by edwinljacobs** [comment]
That is true, this module does not have that functionality. The setting mentioned facilitates that if the filter combination on the category page has a matching landingpage then ajax is not used and the user is sent to the landingpage instead (so normal navigation).

**Comment by koentjeh** [comment]
The attribute landing content still doesn't show with both settings you mentioned enabled and the filter combination matching with the filter attributes on the landing page. It still does an Ajax request instead of loading the landing page, therefore showing the correct filters and filtered products but not the landing page content.

**Comment by koentjeh** [comment]
At first, thank you for your responses. I have managed to find the issue.

Each filter from the layered navigation and the filters in a landings page url will create a unique hash using the createHashForFilters() method in m2-attributelanding/src/model/UrlFinder based on the filters. As mentioned above if the hash of a filter combination is the same as the hash of landings page, the page will be reloaded using normal navigation.

My issue was that the hashes where not the same because the hash was based on a capitalized filter.

Will **not** work
<img width="946" alt="Schermafbeelding 2021-01-11 om 14 51 07" src="https://user-images.githubusercontent.com/6535718/104190508-7ee9c880-541c-11eb-91ac-bb1b9195323d.png">

Does work
<img width="948" alt="Schermafbeelding 2021-01-11 om 14 51 01" src="https://user-images.githubusercontent.com/6535718/104190519-8610d680-541c-11eb-8153-f0f9ad110fe3.png">

The issue was with my configuration but it would be more user friendly if it wouldn't matter whatever capitalization is being used. Would love to hear if you would look further in the issue. If not, this issue can be closed.

**Comment by edwinljacobs** [comment]
Sorry for the late response,

Yeah that is a known issue and I agree that it is not user friendly. If I am not mistaken Tweakwise is considering an api for its "properties" and their values, it is currently not possible to ask tweakwise which values they have for a certain property. When and if this comes around we could render a select for some filters and their values. So there is something on the roadmap for exactly this issue. I dont know when that will be implemented though.

"The issue was with my configuration but it would be more user friendly if it wouldn't matter whatever capitalization is being used. Would love to hear if you would look further in the issue. If not, this issue can be closed." I'll take a look if there is a fix that does not break existing installations.

With kind regards

**Comment by edwinljacobs** [comment]
I checked if we can fix this issue however I cannot guarantee that I do not break existing installations. So this will remain an issue for the near future. I updated the README to at least mention that this is case sensitive. The updated readme will be added to the next release. Perhaps it saves somebody some time in the future. 

The fact that we both spent some time figuring out what was going on is a good indicator that the current behaviour is not user friendly. Thanks for debugging and taking the time to report your findings! For now I'll close the issue

With kind regards
